sudoku/hard/hard.js

```diff
@@ -0,0 +1,145 @@
+import { start, pause, reset } from '../general/stopwatch.js';
+import { addRandom } from './addRandomHard.js';
+import { answers, checkColumnsArr, checkGrid, checkRowsArr } from './checkDublicatesBefore.js';
+import { answersFilled, checkGridFilled } from './checkFunctionsFilled.js';
+// import { start, pause } from './stopwatch.js';
+
+let checkBtn = document.querySelector('.checkHard');
+export let randomBtn = document.querySelector('.addRandom');
+
+///creating 9 divs
+let hard = document.querySelector('.hard');
+for (let i = 0; i < 9; i++) {
+  let div1 = document.createElement('div');
+  div1.classList.add('grid');
+  hard.appendChild(div1);
+}
+// creating 81 divs
+let inner = document.querySelectorAll('.hard>div');
+for (let i = 0; i < 9; i++) {
+  for (let j = 0; j < 9; j++) {
+    let div2 = document.createElement('div');
+    inner[j].appendChild(div2);
+  }
+}
+
+//filling grid with  random numbers
+let grid = document.querySelectorAll('.grid');
+function random() {
+  arr.length = 0;
+  matrix.length = 0;
+  checkRowsArr.length = 0;
+  newmatrix.length = 0;
+  newarr.length = 0;
+  answers.length = 0;
+
+  checkColumnsArr.length = 0;
+
+  for (let i = 0; i < grid.length; i++) {
+    addRandom(grid[i].children);
+  }
+}
+
+//   //emptying grid cells
+checkBtn.addEventListener('click', () => {
+  for (let i = 0; i < grid.length; i++) {
+    for (let j = 0; j < grid.length; j++) {
+      grid[i].children[j].innerHTML = '';
+    }
+  }
+});
+////////calling random
+checkBtn.addEventListener('click', random);
+
+export let matrix = [];
+let arr = [];
+// filing array  with gird numbers
+function fillArray(arr) {
+  for (let i = 0; i <= 8; i++) {
+    for (let j = 0; j <= 8; j++) {
+      arr.push(parseInt(grid[i].children[j].innerHTML));
+    }
+  }
+  return arr;
+}
+
+checkBtn.addEventListener('click', () => {
+  fillArray(arr);
+});
+
+////filing matrix  with gird numbers from array
+
+function split(arr, len) {
+  let i, n;
+
+  (i = 0), (n = arr.length);
+  while (i < n) {
+    matrix.push(arr.slice(i, (i += len)));
```

Comment:
```suggestion
    matrix.push(arr.slice(i, i + len));
    i += len;
```

this way it is more understandable